src/handlers/graphical.rs

```diff
@@ -199,8 +200,8 @@ impl GraphicalReportHandler {
 
         writeln!(f, "{}", textwrap::fill(&diagnostic.to_string(), opts))?;
 
-        if let Some(cause) = diagnostic.source() {
-            let mut cause_iter = Chain::new(cause).peekable();
+        if let Some(cause) = diagnostic.caused_by() {
+            let mut cause_iter = DiagnosticChain::new(cause).peekable();
```

Comment:
I want `#[source]` to continue working for folks who are already using that, or who don't want to opt into `#[caused_by]`. Thoughts? It seems pretty doable.

Reply:
IIUC, `DiagnosticChain` does do this: https://github.com/zkat/miette/pull/165/files/67b27eb94e80e27e4a4addc1a0a1bf03c4ed785f#diff-dd270622a98a63e007d36dd28857130489803df52855385b3150ef332ce4d344R61

It tries to stay on the `Diagnostic` path as long as it can

Reply:
No, that assumes that at the very least the toplevel "source" is a `Diagnostic`, only following the `Error::source` trail _after_ the toplevel one has been resolved. I'm talking about the case where `#[caused_by]` is never defined at all, and `thiserror`'s `#[source]` is used instead (or a manual impl of `Error::source`).

Reply:
You're right! Totally missed that...

Reply:
I pushed a fix which should address this.